### PR TITLE
Course creation fixes

### PR DESCRIPTION
**Pull request by melytc**
This PR solves the following fixes:
- Select user as the default teacher for course creation.
- Correct margin/padding for input fields.
- Input validation done with Semantic UI rules.
- Error messages shown correctly with field highlighting.
- Input legends are now shown as user help (instead of using title attribute).

## Desktop view
Screenshot_2019-04-21 Bluemarket - Register a course

## Mobile view
Screenshot_2019-04-21 Bluemarket - Register a course(1)

### CODE REVIEW

File: resources/views/courses/create.blade.php [hunk 1]
```diff
@@ -11,36 +11,39 @@
...
 				@endforeach
 			</select>
+			<legend>Are you giving this course with another teacher?</legend>
 		</div>
 		<!-- Course semester -->
 		<div class="field">
 			<label for="courseSemester">Course semester</label>
-			<input type="text" name="courseSemester" id="courseSemester" placeholder="Spring/Summer/Fall/Winter and year.  Ex. Fall 2018" title="When is this course taking place?"required>
+			<input type="text" name="courseSemester" id="courseSemester">
+			<legend>When is this course taking place? Spring/Summer/Fall/Winter and year. Ex. Fall 2018</legend>
```
Comment (katiearriagam) [inline resources/views/courses/create.blade.php]: Nit-pick: "e.g. Fall 2018", per the style guide.

File: resources/views/courses/create.blade.php [hunk 2]
```diff
@@ -11,36 +11,39 @@
...
 		</div>
 		<!-- Teacher(s) -->
-		<div class="field" title="Are you giving this course with another teacher?">
+		<div class="field">
 			<label for="teachers">Teacher(s)</label>
-			<select class="ui fluid search dropdown" name="teachers[]" id="teachers" multiple required>
+			<select class="ui fluid search dropdown" name="teachers[]" id="teachers" multiple>
 				@foreach ($teachers as $teacher)
-					<option value={{ $teacher['id'] }}> {{ $teacher['name'] }} </option>
+					<option value={{ $teacher['id'] }} {{ ($teacher['id'] == Auth::user()->id) ? 'selected' : '' }}> {{ $teacher['name'] }} </option>
```
Comment (katiearriagam) [inline resources/views/courses/create.blade.php]: Nit: en un PR pasado, decidimos usar la arrow notación `$teacher->id` 

image

Comment (hecerinc) [reply resources/views/courses/create.blade.php]: Double quote node attributes: (i.e. `value="..."`). Also if you're only getting the ID use `Auth::id()`

File: resources/views/courses/create.blade.php (hunk 1 above)
Comment (hecerinc) [reply resources/views/courses/create.blade.php]: Let's not use the `legend` attribute ([as this is meant for another thing](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/legend)). You can use a `p` here.

File: resources/views/courses/create.blade.php [hunk 3]
```diff
@@ -18,26 +18,26 @@
 			<label for="teachers">Teacher(s)</label>
 			<select class="ui fluid search dropdown" name="teachers[]" id="teachers" multiple>
 				@foreach ($teachers as $teacher)
-					<option value={{ $teacher['id'] }} {{ ($teacher['id'] == Auth::user()->id) ? 'selected' : '' }}> {{ $teacher['name'] }} </option>
+					<option value={{ $teacher->id }} {{ ($teacher->id == Auth::id()) ? "selected" : '' }}> {{ $teacher->name }} </option>
```
Comment (hecerinc) [inline resources/views/courses/create.blade.php]:
```suggestion
<option value="{{ $teacher->id }}" ...></option>
```

Comment (hecerinc) [reply resources/views/courses/create.blade.php]: It's the **`value`** that needs the double quoting, not the `selected`

**Pull request merged**